sqlglot/dialects/snowflake.py

```diff
@@ -115,13 +120,28 @@ class Parser(Parser):
             ),
         }
 
+        def _parse_extract(self):
```

Comment:
this doesn't seem necessary to put into snowflake specifically, it is compatible with the generic parser, can you move this additional functionality into the core parser and remove this custom override?

Reply:
Will do